### ISSUE TITLE
[Optimizely App] variation container allows incorrect variant type to be selected on first render

### ISSUE THREAD

**Issue by LucasAmos** [contentful/apps#3790]
**Describe the bug**
Given that an optimizely experiment has been linked to the contentful environment and 2 or more reference fields have been enabled on a content type. The first time that a new variation container entry is rendered on the screen any enabled content type can be selected from the "Content type" dropdown menu and added to the variation container. The next time that the entry is rendered only the correct content type can be selected.

**To Reproduce**
1. Install the optimizely app
2. Select 2 or more reference fields for a given content type
3. Add a new variation container to an entry
4. Select an optimizely experiment
5. For a variation click "Link an existing entry"
6. All content types will be visible 
7. Select an invalid content type 
8. The content type is added to the variation container
9. Re-render the variation container entry (e.g. refresh the page)
10. Only entries of the valid content type can now be selected despite an invalid content type entry having been added to the variation container

**Expected behavior**
Only entries of the correct content type can be added to the variation container via the "Content type" dropdown menu

**Screenshots**

<img width="893" alt="Screenshot 2023-06-16 at 16 12 35" src="https://github.com/contentful/apps/assets/9055266/2bf8851c-1a4d-4bd8-b5d1-1c2a8a448760">
<img width="1472" alt="Screenshot 2023-06-16 at 16 12 58" src="https://github.com/contentful/apps/assets/9055266/29e81d75-ac24-4e22-906d-974388d0e9f6">
<img width="1472" alt="Screenshot 2023-06-16 at 16 13 29" src="https://github.com/contentful/apps/assets/9055266/cacfb6e0-984b-4651-806f-baf052207dc2">
<img width="1472" alt="Screenshot 2023-06-16 at 16 14 01" src="https://github.com/contentful/apps/assets/9055266/fb8f2e1a-ff3b-47fe-90a2-c57d4ea0566b">
<img width="1472" alt="Screenshot 2023-06-16 at 16 14 28" src="https://github.com/contentful/apps/assets/9055266/64dbef86-50dd-41f0-bb07-d533fe670526">
<img width="1472" alt="Screenshot 2023-06-16 at 16 14 44" src="https://github.com/contentful/apps/assets/9055266/58d2ebb3-b9d5-4bd5-9101-4252d7f1681f">

**Desktop (please complete the following information):**
 - OS: MacOS 13.4
 - Browser Chrome
 - Version 114.0.5735.106 (Official Build) (arm64)

**Additional context**

**Comment by whitelisab** [comment]
Hi @LucasAmos! Thank you for raising this issue. As communicated through our support process, we have captured this issue in our backlog and will prioritize it as soon as possible.

**Comment by whitelisab** [comment]
Just another quick update @LucasAmos. I am going to close this issue and we will continue to communicate updates to you about this bug through our internal support process. Thank you!

**Comment by juancarlosjr97** [comment]
Any update on this ticket? As this has been closed makes following it up is practically impossible.

**Comment by whitelisab** [comment]
@juancarlosjr97 Thanks for reaching out! This ticket has not been prioritized, but I will provide an update when we are working on it.

**Comment by juancarlosjr97** [comment]
@whitelisab... As @LucasAmos mentioned, the issue is the line https://github.com/contentful/apps/blob/ccd11a69d0274a9a35cdc6b0d89f5fd069f41671/apps/optimizely/src/EditorPage/index.js#L86 that does not return the expected response but instead an empty array.

The reason, we think, is happening is because when the entry is created, it opens immediately the drawer and does the fetch of the L86, but as Contentful has just created the entry, the relationship does not exist yet, and if the patch of the parent entry with the newly created entry happens before the fetch data of L86 the array returns correct data, but if it happens after, it returns an empty array as the linked relationship does not exist yet.

Long story short, it requires an await for however long contentful takes to save an empty when a change happens on the UI and does the path.

**Comment by juancarlosjr97** [comment]
I have raised an MR to fix the bug - https://github.com/contentful/apps/pull/4621.

Please, review it.
